### ISSUE TITLE
QoD: start=1489025420000&end=1489026013

### ISSUE THREAD

**Issue by tomwilkie** [weaveworks/cortex#327]
Causes query service to DoS dynamodb

**Comment by Bplotka** [comment]
..which at the end creates massive amount of file descriptors:
![image](https://cloud.githubusercontent.com/assets/6950331/23763887/58e63e56-04f3-11e7-8296-26c2ba7c0c0e.png)
and all of this (amount of requests to DynamoDB) blocks whole node (e.g ingester which was on the same node as querier)
![image](https://cloud.githubusercontent.com/assets/6950331/23763957/a842372a-04f3-11e7-9814-3d4b51f4b4e5.png)

**Comment by Bplotka** [comment]
Restart helps.. simple validation will probably fix the issue.

**Comment by Bplotka** [comment]
Unfortunately this is still valid. I dived into the issue and it seems that the confusion here is because Prometheus wrongly parses my mistaken `start=1489025420000` timestamp. From query perspective it is really start > time problem, but actually underlying int64 overflow I fixed here:  https://github.com/prometheus/prometheus/pull/2501 caused start to be huge negative value.

This means that QoD is still possible and is caused not by start > end, but actually huge range (e.g `start=MinInt64&end=MaxInt64`)
